Makefile

```diff
@@ -162,13 +162,45 @@ quiet = $(if $(V),$($(1)),$(call output,$1,$@);$($(1)))
 this: $(libceed) $(ceed.pc)
 # run 'this' target in parallel
 all:;@$(MAKE) $(MFLAGS) V=$(V) this
+info:
+	$(info ------------------------------------)
+	$(info CC        = $(CC))
+	$(info FC        = $(FC))
+	$(info CPPFLAGS  = $(CPPFLAGS))
+	$(info CFLAGS    = $(value CFLAGS))
+	$(info FFLAGS    = $(value FFLAGS))
+	$(info NVCCFLAGS = $(value NVCCFLAGS))
+	$(info LDFLAGS   = $(value LDFLAGS))
+	$(info LDLIBS    = $(LDLIBS))
+	$(info OPT       = $(OPT))
+	$(info AFLAGS    = $(AFLAGS))
+	$(info ASAN      = $(or $(ASAN),(empty)))
+	$(info V         = $(or $(V),(empty)) [verbose=$(if $(V),on,off)])
+	$(info ------------------------------------)
+	$(info CUDA_DIR  = $(CUDA_DIR))
+	$(info OCCA_DIR  = $(OCCA_DIR))
+	$(info MAGMA_DIR = $(MAGMA_DIR))
+	$(info ------------------------------------)
+	$(info MFEM_DIR  = $(MFEM_DIR))
+	$(info PETSC_DIR = $(PETSC_DIR))
+	$(info ------------------------------------)
+	$(info prefix       = $(prefix))
+	$(info includedir   = $(value includedir))
+	$(info libdir       = $(value libdir))
+	$(info okldir       = $(value okldir))
+	$(info pkgconfigdir = $(value pkgconfigdir))
+	$(info ------------------------------------)
+	@true
+.PHONY: info
 
 $(libceed) : LDFLAGS += $(if $(DARWIN), -install_name @rpath/$(notdir $(libceed)))
 
 libceed.c += $(ref.c)
 libceed.c += $(template.c)
 libceed.c += $(optimized.c)
 ifneq ($(wildcard $(OCCA_DIR)/lib/libocca.*),)
+  OCCA_ON = YES
+  $(info libCEED: OCCA backend enabled [OCCA_DIR=$(OCCA_DIR)])
```

Comment:
I'm not wild about these printing unconditionally (including in `make print-*`, `clean`, `info`, `style`, and the like). Could we instead have a target `info-backends` that is listed as a prerequisite for only specific targets?

Reply:
I'd most like to see the information printed in `make`, `make prove`, `make prove-all`, and `make test` personally. It doesn't make sense for me with many of the others.

Reply:
The reason I wanted these printed all the time is because the auto-detection happens all the time.

I'll make some changes to restrict auto-detection to specific targets, and printing of the auto-detected backends to another (smaller) set of targets.

Reply:
Understood that the detection happens, but printing is just noise if it doesn't affect the target being built. (And many such lines are noise even if they affect the target being built. Perhaps there should be just one line `Using optional backends: OCCA MAGMA`.)